### PR TITLE
fix all clang -Weverything warnings

### PR DESCRIPTION
**Pull request by ydm**
Fix all clang -Weverything warnings

### CODE REVIEW

File: argparse.c [hunk 1]
```diff
@@ -139,9 +147,6 @@ argparse_options_check(const struct argparse_option *options)
         case ARGPARSE_OPT_STRING:
         case ARGPARSE_OPT_GROUP:
             continue;
-        default:
-            fprintf(stderr, "wrong option type: %d", options->type);
-            break;
```
Comment (cofyc) [inline argparse.c]: why this is removed? user may fill in an invalid option type.

File: argparse.c [hunk 2]
```diff
@@ -114,7 +121,8 @@ argparse_getvalue(struct argparse *self, const struct argparse_option *opt,
         if (s[0] != '\0')
             argparse_error(self, opt, "expects a numerical value", flags);
         break;
-    default:
+    case ARGPARSE_OPT_END:
+    case ARGPARSE_OPT_GROUP:
```
Comment (cofyc) [inline argparse.c]: I guess it's better to have a `default` statement here. Even if all enumeration values are covered here, the programmer can still fill arbitrary integer value in the option type field. It's defensive programming.

From my understanding, I don't think it's necessary to fix all warnings emitted with `-Weverything` flag.

Comment (ydm) [reply argparse.c]: You are right, of course, and that's why they're warnings and not errors. It's a pull request, you may disapprove it at will. Anyway, filling arbitrary values to an enum of specific type is a kind of misbehaviour you can't save people from, but I guess it always pays off to be a bit more defensive.

**Pull request closed without merging**